src/contracts.rs

```diff
@@ -21,25 +21,45 @@ use crate::interop::{InteropContext, Interop};
 #[allow(non_snake_case)]
 pub(crate) struct ParamsOfDeploy {
     pub abi: serde_json::Value,
+    pub constructorHeader: Option<serde_json::Value>,
     pub constructorParams: serde_json::Value,
+    pub initParams: Option<serde_json::Value>,
     pub imageBase64: String,
     pub keyPair: Ed25519KeyPair,
+    pub workchainId: i32,
 }
 
-#[derive(Serialize)]
+/// Result of `deploy` function running. Contains address of the contract
 #[allow(non_snake_case)]
-pub(crate) struct ParamsOfGetDeployAddress {
-    pub abi: serde_json::Value,
-    pub imageBase64: String,
-    pub keyPair: Ed25519KeyPair,
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct ResultOfDeploy {
+    pub address: TonAddress,
+    pub alreadyDeployed: bool, 
 }
 
-/// Result of `deploy` and `get_deploy_address` function running. Contains address of the contract
+#[derive(Serialize)]
 #[allow(non_snake_case)]
+pub(crate) struct ParamsOfGetDeployData {
+    pub abi: Option<serde_json::Value>,
+    pub imageBase64: Option<String>,
+    pub initParams: Option<serde_json::Value>,
+    pub publicKeyHex: Ed25519Public,
+    pub workchainId: Option<i32>,
+}
+
 #[derive(Serialize, Deserialize)]
-pub struct ResultOfDeploy {
-    pub address: TonAddress,
-    pub alreadyDeployed: bool, 
+#[allow(non_snake_case)]
+pub(crate) struct ResultOfGetDeployDataCore {
+    pub imageBase64: Option<String>,
+    pub address: Option<String>,
+    pub dataBase64: String,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct ResultOfGetDeployData {
+    pub image: Option<Vec<u8>>,
+    pub address: Option<TonAddress>,
```

Comment:
Why it is an `Option<address>`? When it can be empty?

Reply:
It can be empty if `image` is not provided. We can't calculate address using only data